src/antsibull_core/collection_meta.py

```diff
@@ -75,7 +82,42 @@ def _validate_collection(
         if meta.removal:
             self._validate_removal(collection, meta.removal, f"{prefix} removal ->")
 
-    def validate(self, data: CollectionsMetadata) -> None:
+    def _validate_removal_for_removed(
+        self, collection: str, removal: RemovedRemovalInformation, prefix: str
+    ) -> None:
+        if removal.major_version != self.major_release:
```

Comment:
Should `major_version` not be part of `RemovedRemovalInformation`? Having `removed_collections[NAME].removal.major_version` and `removed_collection[NAME].removed_version` seems duplicative.

Reply:
My idea was that one can simply move the metadata entry for the removed collection to the `removed_collections` list with minimal modifications (the `removal_version` entry needs to be added).

But since that modification is needed anyway, how about replacing `major_version` with a new `version` entry that replaces the `removal_version` one level up? I originally wanted to use an unmodified `RemovalInformation` here, that's why I added `removal_version` one level up, but now that there's a different class it might be easier to do it this way.

Reply:
> My idea was that one can simply move the metadata entry for the removed collection to the `removed_collections` list with minimal modifications (the `removal_version` entry needs to be added).

Yeah, I figured that as well, but then I wondered if that was worth the potential confusion of having the same information twice.


> But since that modification is needed anyway, how about replacing `major_version` with a new `version` entry that replaces the `removal_version` one level up? I originally wanted to use an unmodified `RemovalInformation` here, that's why I added `removal_version` one level up, but now that there's a different class it might be easier to do it this way.

That sounds reasonable.

Reply:
I implemented that in 47593714d43b0fc8f73d6bbe00ab4f2acb511ae3.